skyportal/handlers/api/plan.py

```diff
@@ -0,0 +1,478 @@
+    params["doMovie"] = False
+    params["doObservability"] = True
+    params["do3D"] = False
+
+    params["doFootprint"] = False
+    params["footprint_ra"] = 30.0
+    params["footprint_dec"] = 60.0
+    params["footprint_radius"] = 10.0
+
+    params["airmass"] = airmass
+
+    params["doCommitDatabase"] = True
+    params["doRequestScheduler"] = False
+    params["dateobs"] = dateobs
+    params["doEvent"] = False
+    params["doSkymap"] = False
+    params["doFootprint"] = False
+    params["doDatabase"] = True
+    params["doReferences"] = doReferences
+    params["doUsePrimary"] = doUsePrimary
+    params["doBalanceExposure"] = doBalanceExposure
+    params["doSplit"] = False
+    params["doParallel"] = False
+    params["doUseCatalog"] = False
+
+    params["doMinimalTiling"] = True
+    params["doIterativeTiling"] = False
+    params["galaxies_FoV_sep"] = 1.0
+    params["doMaxTiles"] = doMaxTiles
+    params["max_nb_tiles"] = np.array([max_nb_tiles] * len(filt))
+
+    if params["doEvent"]:
+        params["skymap"], eventinfo = gwemopt.gracedb.get_event(params)
+        params["gpstime"] = eventinfo["gpstime"]
+        event_time = time.Time(params["gpstime"], format='gps', scale='utc')
+        params["dateobs"] = event_time.iso
+    elif params["doSkymap"]:
+        event_time = time.Time(params["gpstime"], format='gps', scale='utc')
+        params["dateobs"] = event_time.iso
+    elif params["doFootprint"]:
+        params["skymap"] = gwemopt.footprint.get_skymap(params)
+        event_time = time.Time(params["gpstime"], format='gps', scale='utc')
+        params["dateobs"] = event_time.iso
+    elif params["doDatabase"]:
+        event_time = time.Time(params["dateobs"], format='datetime', scale='utc')
+        params["gpstime"] = event_time.gps
+    else:
+        raise ValueError(
+            'Need to enable --doEvent, --doFootprint, ' '--doSkymap, or --doDatabase'
+        )
+
+    if tobs is None:
+        now_time = time.Time.now()
+        timediff = now_time.gps - event_time.gps
+        timediff_days = timediff / 86400.0
+        params["Tobs"] = np.array([timediff_days, timediff_days + 1])
+    else:
+        params["Tobs"] = tobs
+
+    params["doSingleExposure"] = True
+    if filterScheduleType == "block":
+        params["doAlternatingFilters"] = True
+    else:
+        params["doAlternatingFilters"] = False
+    params["filters"] = filt
+    params["exposuretimes"] = exposuretimes
+    params["mindiff"] = mindiff
+
+    params["doRASlice"] = doRASlice
+    params["raslice"] = raslice
+
+    params = gwemopt.segments.get_telescope_segments(params)
+    params = gwemopt.utils.params_checker(params)
+
+    return params
+
+
+def gen_structs(params):
+    """Use gwemopt to create observing plan for specified instrument
+    and time window."""
+
+    print('Loading skymap')
+    # Function to read maps
+    map_struct = gwemopt.utils.read_skymap(
+        params, is3D=params["do3D"], map_struct=params['map_struct']
```

Comment:
I don't see `map_struct` anywhere else; how do we know what's supposed to go in there?

Reply:
we currently define it in create_plan... it doesn't need to be set in that dictionary I suppose.